klaytnetl/service/klaytn_contract_service.py

```diff
@@ -45,9 +45,17 @@ def get_function_sighashes(self, bytecode):
                 push4_instructions = [
                     inst for inst in instructions if inst.name == "PUSH4"
                 ]
-                return sorted(
-                    list(set("0x" + inst.operand for inst in push4_instructions))
-                )
+
+                push4_list = list(set("0x" + inst.operand for inst in push4_instructions))
+
+                # Special case when balanceOf(address,uint256) becomes PUSH3 due to optimization
+                # https://github.com/blockchain-etl/ethereum-etl/issues/349#issuecomment-1243352201
+                push3_list_erc_1155 = [
+                    "0x00" + inst.operand for inst in instructions if inst.name == "PUSH3" and inst.operand == "fdd58e"
```

Comment:
Why did you divide the funchash `00fdd582` to `0x00` + `fdd58e`?

Reply:
Since it is `PUSH3`, we need `0x00`, not `0x`

Reply:
I got this.